src/recipebook/recipebook.py

```diff
@@ -473,22 +474,29 @@ def find_changed_recipe_files(root: str, branch="master") -> List[str]:
 
     Returns: List[str]
     """
+
     try:
-        out = subprocess.check_output(["git", "diff", "--name-only",
-                                       "--diff-filter=d", branch, root])
+        cmd = ["git", "rev-parse", "--abbrev-ref", "HEAD"]
+        current_branch = subprocess.check_output(cmd, cwd=root).decode(
+                "utf-8").rstrip()
+
+        cmd = ["git", "diff", "--name-only", "--diff-filter=d", branch]
+        out = subprocess.check_output(cmd, cwd=root)
     except subprocess.CalledProcessError as e:
         log.error(e.output)
         raise e
 
     files = [line.rstrip() for line in out.decode("utf-8").split("\n")]
-    dirs = list(set([os.path.dirname(f) for f in files]))
+    dirs = list(set([Path(f).resolve().parent for f in files]))
     dirs.sort()
 
+    recipe_dir = Path(root).resolve()
     recipe_files = []
     for d in dirs:
-        recipe = os.path.join(d, "meta.yaml")
-        if os.path.isfile(recipe):
-            recipe_files.append(recipe)
+        recipe = d / Path("meta.yaml")
+        if recipe.is_file() and recipe.is_relative_to(recipe_dir):
```

Comment:
`is_relative_to` is new in python 3.9

Reply:
Thanks for spotting. I don't think we have a reason to stay on 3.8, so I'll fix the steup.